src/hooks/use-composed-ref.ts

```
@@ -0,0 +1,40 @@
+    // Safe assignment without MutableRefObject
+    ;(ref as { current: T | null }).current = value
+  }
+}
+
+export const useComposedRef = <T extends HTMLElement>(libRef: React.RefObject<T | null>, userRef: UserRef<T>) => {
+  const prevUserRef = useRef<UserRef<T>>(null)
+
+  return useCallback(
+    (instance: T | null) => {
+      if (libRef && 'current' in libRef) {
+        ;(libRef as { current: T | null }).current = instance
```

ESLint

[error] Error: This value cannot be modified

Modifying component props or hook arguments is not allowed. Consider using a local variable instead.

/home/jailuser/git/src/hooks/use-composed-ref.ts:23:11
  21 |     (instance: T | null) => {
  22 |       if (libRef && 'current' in libRef) {
> 23 |         ;(libRef as { current: T | null }).current = instance
     |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `libRef` cannot be modified
  24 |       }
  25 |
  26 |       if (prevUserRef.current) {

Hint: If this value is a Ref (value returned by `useRef()`), rename the variable to end in "Ref".

(react-hooks/immutability)